File: outbox/poll-based/orderservice/src/main/java/com/varunu28/orderservice/config/RestClientConfig.java
package com.varunu28.orderservice.config;

import org.springframework.beans.factory.annotation.Value;
import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;
import org.springframework.http.HttpHeaders;
import org.springframework.http.MediaType;
import org.springframework.web.client.RestClient;

@Configuration
public class RestClientConfig {

    @Value("${notification.service.url}")
    private String notificationServiceUrl;

    @Bean
    public RestClient notificationServiceRestClient() {
        return RestClient.builder()
            .baseUrl(notificationServiceUrl)
            .defaultHeader(HttpHeaders.CONTENT_TYPE, MediaType.APPLICATION_JSON_VALUE)
            .build();
    }
}

File: saga/order-service/src/main/kotlin/com/saga/orderservice/saga/RestClientConfig.kt
package com.saga.orderservice.saga

import org.springframework.beans.factory.annotation.Qualifier
import org.springframework.beans.factory.annotation.Value
import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration
import org.springframework.web.client.RestClient

@Configuration
class RestClientConfig {

    @Bean
    @Qualifier("paymentServiceRestClient")
    fun paymentServiceRestClient(@Value($$"${payment.service.url}") baseUrl: String): RestClient {
        return RestClient.builder()
            .baseUrl(baseUrl)
            .build()
    }

    @Bean
    @Qualifier("inventoryServiceRestClient")
    fun inventoryServiceRestClient(@Value($$"${inventory.service.url}") baseUrl: String): RestClient {
        return RestClient.builder()
            .baseUrl(baseUrl)
            .build()
    }
}